akka-stream-tests/src/test/scala/akka/stream/impl/fusing/InterpreterSpec.scala

```diff
@@ -569,39 +569,69 @@ class InterpreterSpec extends AkkaSpec with GraphInterpreterSpecKit {
 
   }
 
-  private[akka] case class Doubler[T]() extends PushPullStage[T, T] {
-    var oneMore: Boolean = false
-    var lastElem: T = _
-
-    override def onPush(elem: T, ctx: Context[T]): SyncDirective = {
-      lastElem = elem
-      oneMore = true
-      ctx.push(elem)
-    }
+  private[akka] final case class Doubler[T]() extends GraphStage[FlowShape[T, T]] {
```

Comment:
> > Actually, as GraphStages are reusable it could now be an object
> Yes, I could do that. But the parameter type will be missing. Am I right?
 
I don't think it's worth it in this case - this is just a test utility class.

In general though it is a nice idea, and we do work around the lack of type parameter using similar patterns to this one:

```scala
object Anything extends GraphStage[FlowShape[Any, Any]]  {}
def anything[T] = Anything.asInstanceOf[GraphStage[FlowShape[T, T]]]
```

which works, and we saved an allocation :)